tests/testing_config/custom_components/test/vacuum.py

```diff
@@ -0,0 +1,83 @@
+"""Provide a mock vacuum platform.
+
+Call init before using it in your tests to ensure clean test data.
+"""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.vacuum import (
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    StateVacuumEntity,
+    VacuumEntityFeature,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockEntity
+
+ENTITIES = []
+
+
+def init(empty=False):
+    """Initialize the platform with entities."""
+    global ENTITIES
+    ENTITIES = [] if empty else [MockVacuum()]
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config, async_add_entities_callback, discovery_info=None
+):
+    """Return mock entities."""
+    async_add_entities_callback(ENTITIES)
+
+
+class MockVacuum(MockEntity, StateVacuumEntity):
```

Comment:
I suppose we can put this in `__init__.py` instead as the above part is not used?

Reply:
I wasn't sure as it isn't used in the weather part either. I just figured it was meant to make testing custom components easier and it was used somewhere in that sense? I'm not sure. Happy to move to init if you think that makes more sense

Reply:
If it's not used in `WeatherEntity` tests anymore it was forgotten to be cleaned up. I'll check that

Reply:
> If it's not used in `WeatherEntity` tests anymore it was forgotten to be cleaned up. I'll check that

It also could just be that vscode is stupid and didn't find the actual usages.

But I have moved to init.